### DISCUSSION TITLE
too many hreflang generation

### DISCUSSION THREAD

**Discussion by AlexyOd** [nuxt-modules/i18n#1303] [Q&A]
Hi i can't finde answer on documentation

the plugin create 4(!) hreflang !! 
How to fix this ??? 

` locales: [
          {
            name: "lang",
            code: "ua",
            iso: "ua-FR",
            file: "ua-FR.js"
          },
          {
            name: "lang2",
            code: "fr",
            iso: "fr-FR",
            file: "fr-FR.js"
          }
        ],`

and i got in header 
`
link data-n-head="ssr" data-hid="alternate-hreflang-ua" rel="alternate" href="http://web.work/" hreflang="ua"
link data-n-head="ssr" data-hid="alternate-hreflang-ua-FR" rel="alternate" href="http://web.work/" hreflang="ua-FR"
link data-n-head="ssr" data-hid="alternate-hreflang-fr" rel="alternate" href="http://web.work/ua" hreflang="fr"
link data-n-head="ssr" data-hid="alternate-hreflang-fr-FR" rel="alternate" href="http://web.work/ua" hreflang="fr-FR"
`

but i need only 
        link data-n-head="ssr" data-hid="alternate-hreflang-ua-FR" rel="alternate" href="http://web.work/" hreflang="ua-FR"
link data-n-head="ssr" data-hid="alternate-hreflang-fr-FR" rel="alternate" href="http://web.work/ua" hreflang="fr-FR"

**Comment by rchl** [comment]
Those are catch-all locales and it works as designed and is explained at https://i18n.nuxtjs.org/seo/#feature-details.

**Comment by AlexyOd** [comment]
there is no answer there

**Comment by rchl** [comment]
The documention has a link to https://support.google.com/webmasters/answer/189077 that specifies that it's a recommended practice to include those catchall locales (search for "catchall" on that page).

**Comment by AlexyOd** [comment]
so, 4 hreflang is good ? and "works as designed and is explained at https://i18n.nuxtjs.org/seo/#feature-details." ??? 
![image](https://user-images.githubusercontent.com/13238444/134680481-9e65cb1f-50c0-4047-8a2e-abc3d6bd0e31.png)

**Comment by rchl** [comment]
Did you even read my last comment?

**Comment by AlexyOd** [comment]
yes

**Comment by AlexyOd** [comment]
i stell can't remove hreflang="en"

**Comment by rchl** [comment]
What's the issue you are trying to fix? You just don't like it being there? Then don't use `$nuxtI18nHead` and create your own meta from scratch.
As I said, it's by design and recommended by google.

**Reply by provokcx** [reply]
I have the same problem. I think it would be a great idea to make this feature optional. And better explain in the documentation why people should do this.

In specific cases, this feature enabled by default could cause duplicated content.

See the [head-meta](https://github.com/nuxt-community/i18n-module/blob/d5dea9c6a047e5e25a13904c3a2cb8ffa4fdf089/src/templates/head-meta.js) file at line 84.

```js
if (language && region && (locale.isCatchallLocale || !localeMap.has(language))) {
  localeMap.set(language, locale)
}
```

This specific test won't allow us to even disable the feature.

For now, I have to use a hack to disable this feature. Please consider making it optional.

@rchl @AlexyOd @atinux @danielroe @pi0 At the macro level, what do you think about this?

**Reply by AlexyOd** [reply]
that is interesting solution.

**Reply by reslear** [reply]
new link:

https://github.com/intlify/routing/blob/8d22f26c9398c716efe7ed1b6c3d1a78d9b0baaa/packages/vue-i18n-routing/src/compatibles/head.ts#L101-L103

@provoks-land what do you hack use? 


<img width="383" alt="Screenshot 2023-12-06 at 05 15 15" src="https://github.com/nuxt-modules/i18n/assets/12596485/0ea2d869-69bc-4359-9aed-24ac78408f11">

@BobbieGoede could help with that?

**Reply by provokcx** [reply]
@reslear 
I didn't understand your question.
Can you rephrase it?

**Reply by reslear** [reply]
Oops, sorry, as you said, what hack can be used to disable this feature?

**Reply by reslear** [reply]
hi @kazupon please advise please 🙏
because that's what generates sitemap same https://github.com/harlan-zw/nuxt-simple-sitemap by @harlan-zw

**Reply by BobbieGoede** [reply]
Please open a new issue or discussion with a reproduction and your desired scenario, this discussion is a few years old
